pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -202,6 +202,10 @@ type NetworkPolicyController struct {
 	// concurrent access during updates to the internal NetworkPolicy object.
 	internalNetworkPolicyMutex sync.RWMutex
 
+	// internalGroupMutex protects the internalGroupStore from current access
```

Comment:
concurrent.

Why the mutex is needed? `internalGroupStore` itself is thread-safe. You may want to prevent race condition between the event handlers and syncWorkers but the critical sections in this commit only protects the store operations, which is duplicate with the lock in the store itself.

Reply:
Thanks for the reminder. Removed the group mutex